### PR TITLE
build: support OSX arm64

### LINKED ISSUE #71: build Wolf on OSX arm64
improve and build Wolf on arm64 OSX

### PR DESCRIPTION
**Pull request by PooyaEimandar**
## Pull Request Checklist

What kind of change does this pull request introduce?

- [x] Build: Changes that affect the build system or external dependencies
- [ ] CI: Changes to our CI configuration files and scripts
- [ ] Docs: Documentation only changes
- [ ] Feat: A new feature
- [ ] Fix: A bug fix
- [ ] Perf: A code change that improves performance
- [ ] Refactor: A code change that neither fixes a bug nor adds a feature
- [ ] Test: Adding missing tests or correcting existing tests

## Does this Pull Request introduce a breaking change?
- [ ] Yes
- [x] No

## Other information

Closes #71

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -52,6 +52,22 @@
...
+            "generator": "Ninja",
+            "architecture": {
+                "value": "arm64",
+                "strategy": "external"
+            },
+            "cacheVariables": {
+                "CMAKE_C_COMPILER": "clang",
+                "CMAKE_CXX_COMPILER": "clang++",
+                "CMAKE_CONFIGURATION_TYPES": "Debug;Release",
+                "LIBRARY_TYPE": "STATIC",
```
Comment (cthulhu-irl) [inline CMakePresets.json:67]: please use `BUILD_SHARED_LIBS=OFF` to state static build. it's also off by default.

**Review by cthulhu-irl — CHANGES_REQUESTED** [review]
would you also add OSX to cmake variable `DESKTOP` and/or update feature options accordingly so features supporting OSX would be built too?

File: CMakePresets.json [hunk 2]
```diff
@@ -49,16 +50,35 @@
                 "CMAKE_CXX_COMPILER": "g++",
                 "CMAKE_CONFIGURATION_TYPES": "Debug;Release",
                 "LIBRARY_TYPE": "SHARED",
+                "BUILD_SHARED_LIBS": "OFF",
```
[the remark this answers is not in the document]
Comment (PooyaEimandar) [reply CMakePresets.json:53]: Thanks, fixed.

File: CMakeLists.txt [hunk 3]
```diff
@@ -49,8 +50,8 @@ wolf_feature_option(WOLF_STREAM_RIST "Enable RIST streaming protocol" OFF)
 wolf_feature_option(WOLF_STREAM_WEBRTC "Enable webRTC" OFF)
 
 # system modules
-wolf_feature_option(WOLF_SYSTEM_GAMEPAD_CLIENT "Enable gamepad input handling" WOLF_FEATURES_ALL AND NOT LINUX AND NOT EMSCRIPTEN)
-wolf_feature_option(WOLF_SYSTEM_GAMEPAD_VIRTUAL "Enable virtual gamepad simulator" OFF)
+wolf_feature_option(WOLF_SYSTEM_GAMEPAD_CLIENT "Enable gamepad input handling" WIN64 AND NOT EMSCRIPTEN)
+wolf_feature_option(WOLF_SYSTEM_GAMEPAD_VIRTUAL "Enable virtual gamepad simulator" OFF) #WIN64 AND NOT EMSCRIPTEN
```
[the remark this answers is not in the document]
Comment (PooyaEimandar) [reply CMakeLists.txt]: Thanks, fixed.

**Review by cthulhu-irl — APPROVED** [review]

**Pull request merged**